kratos/processes/time_averaging_process.cpp

```diff
@@ -0,0 +1,292 @@
+{
+    KRATOS_TRY
+
+    if (!mrModel.HasModelPart(mModelPartName))
+    {
+        const std::vector<std::string>& r_model_part_names = mrModel.GetModelPartNames();
+
+        std::string msg;
+        msg = mrModel.Info() + " doesn't have " + mModelPartName +
+              ". Available model parts are: \n";
+        for (std::string model_part_name : r_model_part_names)
+            msg += "     " + model_part_name + "\n";
+
+        KRATOS_ERROR << msg;
+    }
+
+    const ModelPart& r_model_part = mrModel.GetModelPart(mModelPartName);
+
+    for (const std::string& variable_name : mVariableNamesList)
+    {
+        if (KratosComponents<Variable<double>>::Has(variable_name))
+        {
+            const Variable<double>& r_variable =
+                KratosComponents<Variable<double>>::Get(variable_name);
+            KRATOS_ERROR_IF(!r_model_part.HasNodalSolutionStepVariable(r_variable))
+                << r_model_part.Name() << " doesn't have "
+                << r_variable.Name() << " in NodalSolutionStepDataContainer. Please add it as a SolutionStepVariable.";
+        }
+        else if (KratosComponents<Variable<array_1d<double, 3>>>::Has(variable_name))
+        {
+            const Variable<array_1d<double, 3>>& r_variable =
+                KratosComponents<Variable<array_1d<double, 3>>>::Get(variable_name);
+            KRATOS_ERROR_IF(!r_model_part.HasNodalSolutionStepVariable(r_variable))
+                << r_model_part.Name() << " doesn't have "
+                << r_variable.Name() << " in NodalSolutionStepDataContainer. Please add it as a SolutionStepVariable.";
+        }
+        else
+        {
+            KRATOS_ERROR << "Variable " << variable_name << " not found in the scalar and 3d variables list of Kratos.\n";
+        }
+    }
+
+    return 0;
+
+    KRATOS_CATCH("");
+}
+
+void TimeAveragingProcess::ExecuteInitialize()
+{
+    KRATOS_TRY
+
+    ModelPart& r_model_part = mrModel.GetModelPart(mModelPartName);
+    Communicator& r_communicator = r_model_part.GetCommunicator();
+    ModelPart::NodesContainerType& r_nodes = r_communicator.LocalMesh().Nodes();
+
+    mCurrentTime = 0.0;
+
+    std::stringstream msg;
+
+    msg << "Initialized non-historical";
+
+    for (const std::string& variable_name : mVariableNamesList)
+    {
+        if (KratosComponents<Variable<double>>::Has(variable_name))
+        {
+            const Variable<double>& r_variable =
+                KratosComponents<Variable<double>>::Get(variable_name);
+            this->InitializeTimeAveragedQuantity(r_nodes, r_variable);
+        }
+        else if (KratosComponents<Variable<array_1d<double, 3>>>::Has(variable_name))
+        {
+            const Variable<array_1d<double, 3>>& r_variable =
+                KratosComponents<Variable<array_1d<double, 3>>>::Get(variable_name);
+            this->InitializeTimeAveragedQuantity(r_nodes, r_variable);
+        }
+        msg << " " << variable_name << ",";
+    }
+
+    msg.seekp(-1, msg.cur);
```

Comment:
can you restructure your output in order to avoid this?

Reply:
done :)